### PR TITLE
Add docs

### PR DESCRIPTION
**Pull request by maryx**
Resolves:
#48 cleaning up import order/constructors
#63 documentation

### CODE REVIEW

File: unit_converter/unit_converter/lib/api.dart [hunk 1]
```diff
@@ -18,22 +18,22 @@ const apiCategory = {
...
-/// exchange rate.
+/// We have set up an API that retrieves a list of currencies and their current
+/// exchange rate (mock data).
 ///   GET /currency: get a list of currencies
 ///   GET /currency/convert: get conversion from one currency amount to another
 class Api {
-  // We use the `dart:io` HttpClient. More details: https://flutter.io/networking/
-  final httpClient = HttpClient();
+  /// We use the `dart:io` HttpClient. More details: https://flutter.io/networking/
+  final HttpClient httpClient = HttpClient();
```
Comment (maryx) [inline unit_converter/unit_converter/lib/api.dart]: Just confirming, for class variables, we do want to note the type. But for local vars, we do not?

File: unit_converter/unit_converter/lib/unit_converter.dart [hunk 2]
```diff
@@ -167,6 +213,12 @@ class _UnitConverterState extends State<UnitConverter> {
       ));
     }
 
+    if (_inputValue != null) {
+      setState(() {
+        _updateConversion();
```
Comment (maryx) [inline unit_converter/unit_converter/lib/unit_converter.dart]: If I have 3 consecutive if-statements that are unrelated to each other (as here, i only added one in this PR so you'll have to expand to see the other two) and all of them contain a setState(), should I have 3 setState() calls, or one at the end (after all the if statements)?

Comment (yjbanov) [reply unit_converter/unit_converter/lib/unit_converter.dart]: `setState` is not called during build, but in callbacks to gestures (and other external events), so it should not affect frame build time. I'd go with the first option too.

File: unit_converter/unit_converter/lib/api.dart (hunk 1 above)
Comment (yjbanov) [reply unit_converter/unit_converter/lib/api.dart]: I've developed a habit of specifying types in fields, mostly because Flutter's internal style guide enforces it. So I'm the wrong person to ask. However, technically, a `final` field's type is determined by the expression on the right-hand side (because it cannot change to anything else after initialization). If there's a reason to put a type there, it would be purely for readability/personal preference reasons.

File: unit_converter/unit_converter/lib/api.dart [hunk 3]
```diff
@@ -18,22 +18,22 @@ const apiCategory = {
...
   /// The API endpoint we want to hit.
   ///
   /// This API doesn't have a key but often, APIs do require authentication
-  final url = 'flutter.udacity.com';
+  final String url = 'flutter.udacity.com';
 
   /// Gets all the units and conversion rates for a given category.
   ///
-  /// [category] is the category from which to retrieve units.
+  /// `category` is the [Category] from which to retrieve units.
```
Comment (yjbanov) [inline unit_converter/unit_converter/lib/api.dart]: To use backticks to refer to parameters is a Flutter-only quirk. Neither IDEs nor dartdoc will understand it. I'd encourage that you use square brackets (which _are_ understood by IDEs and dartdocs) until dartdoc supports something else.

File: unit_converter/unit_converter/lib/api.dart (hunk 1 above)
Comment (maryx) [reply unit_converter/unit_converter/lib/api.dart]: Thanks!

File: unit_converter/unit_converter/lib/unit_converter.dart (hunk 2 above)
Comment (maryx) [reply unit_converter/unit_converter/lib/unit_converter.dart]: Great, thanks!

File: unit_converter/unit_converter/lib/api.dart (hunk 3 above)
Comment (maryx) [reply unit_converter/unit_converter/lib/api.dart]: Thanks!

**Comment by maryx** [comment]
Merging this as licensing has been moved to https://github.com/flutter/udacity-course/pull/88
